src/main/java/com/harragan/battleshipsboot/controllers/exceptionhandling/ExceptionHandler.kt

```diff
@@ -0,0 +1,26 @@
+package com.harragan.battleshipsboot.controllers.exceptionhandling
+
+import com.harragan.battleshipsboot.service.exceptions.IllegalBoardPlacementException
+import com.harragan.battleshipsboot.service.exceptions.IllegalGameStartException
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.context.request.WebRequest
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler
+
+@ControllerAdvice
+class ExceptionHandler : ResponseEntityExceptionHandler() {
+
+    @ExceptionHandler(IllegalBoardPlacementException::class)
+    fun handleBoardException(exception: IllegalBoardPlacementException
+                             , webRequest: WebRequest): ResponseEntity<String> {
+        return ResponseEntity(exception.message!!, HttpStatus.BAD_REQUEST)
```

Comment:
Add `@ResponseBody` and `@ResponseStatus(BAD_REQUEST)` to the function signature and return the exception's message directly
Could also use `=` here instead of a block with `return`

Reply:
"Could also use = here instead of a block with return"
I am not too sure what you mean - please could you provide an example?